src/components/comment-section/CommentSection.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import { Comment } from '../../types/types';
+import './CommentSection.css';
+
+type CommentSectionProps = {
+  currentUserId: number;
+  currentUserName: string;
+  initialComments: Comment[] | undefined;
+  onCommentsChange: (comments: Comment[]) => void;
+};
+
+const CommentSection: React.FC<CommentSectionProps> = ({
+  currentUserId,
+  currentUserName,
+  initialComments,
+  onCommentsChange,
+}) => {
+  const [comments, setComments] = useState<Comment[] | undefined>(
```

Comment:
you do not need a state for comments in this comments section component, since you already have state for comments in parent component and you are passing it via props
so when you trigger Send comments, you already update state in parent component, and that state is sent via props again.. so please remove this redudant state